File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();

        Map<Character, Character> map = new HashMap<>();
        map.put('(', ')');
        map.put('{', '}');
        map.put('[', ']');

        for (int i = 0; i < s.length(); i++) {
          char ch = s.charAt(i);
          if (map.containsKey(ch)) {
            stack.push(ch);
          } else if (stack.isEmpty() || ch != map.get(stack.pop())) {
            return false;
          }
        }
        
        return stack.isEmpty();
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        parentheses = dict(("()", "[]", "{}"))
        stack = []
        print(parentheses)
        for element in s:
            if element in "([{":
                # If open parentheses are present, append it to stack
                stack.append(element)
            elif len(stack) == 0 or element != parentheses[stack.pop()]:
                # If the character is closing parentheses, check that the same type opening parentheses is being pushed to the stack or not
                return False

        return len(stack) == 0